### DISCUSSION TITLE
Key random value

### DISCUSSION THREAD

**Discussion by CROprogrammer** [sngular/kloadgen#392] [Q&A] [answered]
Hi,

is it possible for KLoadGen to generate random message key for every Kafka message istead of defining one for every message in KLG - Simple Key Config?

**Comment by AdrianLagartera** [comment, marked as answer]
Well Jmeter have a native function to do that https://jmeter.apache.org/usermanual/functions.html#__RandomString 
We will include this information in the [Producer Configuration](https://github.com/sngular/kloadgen/wiki/producer-configuration) to make it easier for people who don't know all the functionality of Jmeter

**Reply by CROprogrammer** [reply]
Amazing thaks :D

But I've noticed that when I use that function in Simple Key Config

![image](https://github.com/sngular/kloadgen/assets/73464772/6a450899-8909-4776-af47-218282b65911)

It returns the same string everytime. I've configured producer so it produces message every 3 seconds, but everytime, the key is the same.

![image](https://github.com/sngular/kloadgen/assets/73464772/94569771-611c-4764-89a3-c89b81c006dc)

Any idea why?

**Reply by AdrianLagartera** [reply]
Because the key is generated at JMeter level, so doesn't matter the time between message, I am not sure if using more than 1 thread that could change, but with my current knowledge of the JMeter I think not, so... you need run every time you want a random key

**Reply by CROprogrammer** [reply]
Yeah multiple threads did the job, thanks
